cmd/crl-checker/main.go

```diff
@@ -79,18 +83,32 @@ func main() {
 	cmd.FailOnError(err, "Parsing age limit")
 
 	errCount := 0
-	for _, url := range urls {
-		crl, err := downloadShard(url)
+	for _, u := range urls {
+		crl, err := downloadShard(u)
 		if err != nil {
 			errCount += 1
-			logger.Errf("fetching CRL %q failed: %s", url, err)
+			logger.Errf("fetching CRL %q failed: %s", u, err)
 			continue
 		}
 
+		if *save {
+			parsedURL, err := url.Parse(u)
+			if err != nil {
+				logger.Errf("parsing url: %s", err)
+				continue
+			}
+			filename := fmt.Sprintf("%s%s", parsedURL.Host, strings.Replace(parsedURL.Path, "/", "_", -1))
```

Comment:
I think `strings.ReplaceAll()` is equivalent to `strings.Replace(..., -1)`, and easier for the reader to understand.